Monika-After-Story/MonikaModDev
# MASButtonDisplayable Enhancements and fixes

Includes fix from #5579 but obsoletes all other changes from that PR, so only merging that change here. The other PR will be closed.

# Key Changes
* From #5579 - fix button backs rendering to entire canvas in MASButtonDisplayable.render
* adds two new functions for button creation
    * `MASButtonDisplayable.create_st` - creates a button where the idle/hover/disable text is all the same 
    * `MASButtonDisplayable.create_stb` - creates a button where the idle/hover/disable text is all the same and the backgrounds use `choice_button`'s `background` frames if possible 
* all instances of MASButtonDisplayable now use grey text when disabled.

### API additions
* `mas_getPropFromStyle` - retrieves value of a property from a style. Use full names for a style (not style prefix). This obtains the actual value used by the style, so making any changes to this will affect anything using the style. Duplicate the object if you need something specific (duplicating is non-trivial so its up to the caller)
* `mas_prefixFrame` - extracts an image path from a frame and applies prefix to it using substitute, then builds a new frame with that image path. Only copies over image and left/top/right/bottom props (borders).

# Testing
* verify correct button behavior and appearance (grey text when disabled) in the following places:
    * chess
        * save - idle/hover/disabled (becomes not disabled after reaches the user's 5/6th turn)
        * give up - idle/hover/disabled (disabled during Monika's turn)
        * done - idle/hover
    * updater menu
        * update - idle/hover/disabled (disabled while checking for update, enabled when update found. could trick this by changing your version to a previous one)
        * cancel - idle/hover
        * retry - idle/hover/disabled (only appears if the first update check fails because of internet problems. becomes disabled while retrying. might be possible to trigger by killing internet connection while first update check runs)
        * Ok - idle/hover - appears if game is up-to-date
    * graphics menu
        * return - idle/hover
        * all the other buttons - idle/hover/disabled (disabled if they are the current renderer)
    * piano (normal mode)
        * config - idle/hover/disabled (disabled when a song has been completed)
        * quit - idle/hover
    * piano (config mode)
        * done - idle/hover/disabled (disabled while configuring a key)
        * cancel - idle/hover/disabled (disabled while **not** configuring a key)
        * reset all - idle/hover/disabled (appears while **not** configuring a key, disabled if no keyboard configuration exists)
        * reset - idle/hover/disabled (appears while configuring a key, disabled if the key has not been configured)
